content/en/docs/concepts/services-networking/network-policies.md

```diff
@@ -312,6 +312,50 @@ The value of the label is the namespace name.
 While NetworkPolicy cannot target a namespace by its name with some object field, you can use the
 standardized label to target a specific namespace.
 
+## Pod lifecycle
+
+{{< note >}}
+The following applies to clusters with a conformant networking plugin and a conformant implementation of
+NetworkPolicy.
+{{< /note >}}
+
+When a new NetworkPolicy object is created, it may take some for a network plugin
+to handle the new object. If the pod that is affected by a NetworkPolicy
+is created before the network plugin has completed NetworkPolicy handling,
+that pod may be started unprotected, and isolation rules will be applied when 
+the NetworkPolicy handling is completed.
+
+Once the NetworkPolicy is handled by a network plugin,
+
+1. All newly created pods affected by a given NetworkPolicy will be isolated before 
+they are started.
+Implementations of NetworkPolicy must ensure that filtering is effective throughout
+the Pod lifecycle, even from the very first instant that any container in that Pod is started.
+Because they are applied at Pod level, NetworkPolicies apply equally to init containers,
+sidecar containers, and regular containers.
+
+2. Allow rules will be applied eventually after the isolation rules (that includes applying both at the same time).
+In the worst case, a newly created pod may have no network connectivity at all, if
+isolation rules were already applied, but no allow rules were applied yet.
+
+Every created NetworkPolicy will be handled by a network plugin eventually, but there is no
+way to tell from the Kubernetes API when exactly that happens. 
+The only way to check if the network policy is applied to a pod, is to run a connectivity check
+from that pod.
+
+The pods must be resilient against being started up with different network
+connectivity than expected. If the pod needs to make sure it has expected network
+connectivity before being started, you can use an [init container](/docs/concepts/workloads/pods/init-containers/)
+that will run required connectivity checks before starting the app containers.
```

Comment:
"...before letting the app containers start." or "...before kubelet starts the app containers".